### PR TITLE
feat: Install esbuild deps with production flag

### PR DESCRIPTION
**Pull request by mildaniel**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws_lambda_builders/workflows/nodejs_npm_esbuild/workflow.py [hunk 1]
```diff
@@ -188,7 +192,7 @@ def _accelerate_workflow_actions(
                     actions += esbuild_no_deps
             else:
                 # Invalid workflow, can't have no dependency dir and no installation
-                raise EsbuildExecutionError(message="Lambda Builders encountered and invalid workflow")
+                raise EsbuildExecutionError(message="Lambda Builders encountered an invalid workflow")
```
Comment (jfuss) [inline aws_lambda_builders/workflows/nodejs_npm_esbuild/workflow.py:195]: Should this error be more specific? The comment has more context which could be useful to help customers understand why the build failed. As a customer "invalid workflow" doesn't really tell me much.

File: aws_lambda_builders/workflows/nodejs_npm_esbuild/workflow.py [hunk 2]
```diff
@@ -127,7 +127,11 @@ def actions_with_bundler(
         )
 
         install_action = NodejsNpmWorkflow.get_install_action(
-            source_dir, scratch_dir, subprocess_npm, osutils, self.options, is_production=False
+            source_dir,
+            scratch_dir,
+            subprocess_npm,
+            osutils,
+            self.options,
```
Comment (jfuss) [inline aws_lambda_builders/workflows/nodejs_npm_esbuild/workflow.py:134]: I assume `is_production` defaults to `True`?

Is there any reason a customer may want `False`? If so, how would a customer do that?

**Review by jfuss — CHANGES_REQUESTED** [review]

File: aws_lambda_builders/workflows/nodejs_npm_esbuild/workflow.py (hunk 2 above)
Comment (mildaniel) [reply aws_lambda_builders/workflows/nodejs_npm_esbuild/workflow.py:134]: Yes, it defaults to true. We don't support any way for customers to set it to False, and we actually don't use it as False anywhere except here so I'm just going to remove that parameter. Setting `production=False` means that development dependencies will also be installed which adds a ton of bloat and install time that customers likely won't want packaged with their Lambda. Before, we expected customers to include `esbuild` as a dev dependency. Now, we expect it to be installed on their PATH or as a normal dependency. This is the behaviour we currently have with our Node.js workflow and are making esbuild consistent.

**Review by jfuss — APPROVED** [review]

File: aws_lambda_builders/workflows/nodejs_npm/workflow.py [hunk 3]
```diff
@@ -180,4 +180,4 @@ def get_install_action(source_dir, artifacts_dir, subprocess_npm, osutils, build
         if (osutils.file_exists(lockfile_path) or osutils.file_exists(shrinkwrap_path)) and npm_ci_option:
             return NodejsNpmCIAction(artifacts_dir, subprocess_npm=subprocess_npm)
 
-        return NodejsNpmInstallAction(artifacts_dir, subprocess_npm=subprocess_npm, is_production=is_production)
+        return NodejsNpmInstallAction(artifacts_dir, subprocess_npm=subprocess_npm, is_production=True)
```
Comment (mildaniel) [inline aws_lambda_builders/workflows/nodejs_npm/workflow.py:183]: We can just remove this parameter.

**Review by lucashuy — APPROVED** [review]

**Pull request merged**